scst-store/use-external-database.hbs.md

```diff
@@ -1,43 +1,56 @@
+   2. (Optional if TLS is enabled and certificate is self signed) CA Certificate that signed the certificate
+   3. Database Credentials
+   4. Name of the datbase to use for AMR/MDS
 
-## Set up certificate and configuration
+- Connectivity to the Postgres datbase is avaiable from the AMR and MDS services
+- If migrating from the internal to an external database, be sure to backup the data before proceeding
 
-1. Create a security group to allow inbound connections from the cluster to the PostgreSQL database.
+## Configuring AMR and MDS to use the external database
 
-2. Retrieve the corresponding CA Certificate that signed the PostgreSQL TLS Certificate.
-
-3. In the `metadata-store-values.yaml` fill the following settings:
+1. Update the TAP values for your deployment to include the following configurations:
 
    ```yaml
-   db_host: "<DB Instance Endpoint>"
-   db_user: "<Master Username>"
-   db_password: "<Master Password>"
-   db_name: "<Database Name>"
-   db_port: "5432"
-   db_sslmode: "verify-full"
-   db_max_open_conns: 10
-   db_max_idle_conns: 100
-   db_conn_max_lifetime: 60
-   db_ca_certificate: |
-      <Corresponding CA Certification>
-      ...
-      ...
-      ...
-   deploy_internal_db: "false"
+   metadata-store: 
+     db_host: "<DB Instance Endpoint>"
+     db_user: "<Master Username>"
+     db_password: "<Master Password>"
+     db_name: "<Database Name>"
+     # SEE NOTE BELOW
+     # Disable the internal database deployment
+     deploy_internal_db: "false"
+     db_port: "5432"
+     db_max_open_conns: 10
+     db_max_idle_conns: 100
+     db_conn_max_lifetime: 60
+     # If TLS is enabled on Postgres Instance
+     db_sslmode: "verify-full"
+     # If TLS Certificate is self-signed
+     db_ca_certificate: |
+       <Corresponding CA Certification>
+       ...
+       ...
+       ...
    ```
+   > **Note** If you initially deployed TAP with an internal datbase and are migrating to an external database, be aware setting `deploy_internal_db` to `false,`  will remove the internal instance of PostgreSQL.  Be sure to backup and migrate your data to the database prior to setting this value to false or it may result in data loss.
 
-> **Note** If `deploy_internal_db` is set to `false,` an instance of PostgreSQL is not deployed in the cluster.
+2. Apply the new configuration
+   ```console
+   tanzu package installed update tap -p tap.tanzu.vmware.com -v {{ vars.tap_version }}  --values-file tap-values.yaml -n tap-install
+   ```
+3. (Optional) If you are migrating from the internal database to an external database, the reconcilation above may fail.  In order for it to succeed, you must manually delete the secret that cert-manager created and cycle the AMR and MDS services so that the new secret can be picked up.
+   ```console
+   kubectl delete secret -n metadata-store postgres-db-tls-cert
+   kubectl rollout restart -n metadata-store deployment metadata-store-app amr-graphql-app
```

Comment:
GraphQL was in alpha before 1.7 and I think hidden behind a feature flag.

So mentioning amr-graphql-app is probably applicable only in 1.7, and not for 1.6 and 1.5.

Reply:
Good call!  I'll just merge this in to 1.7.

Reply:
Actually - scratch that.  I'll merge this to get 1.5 and 1.6, then submit a second one to add amr-graphql-app for just 1.7.